plugin.php

```diff
@@ -5,275 +5,658 @@
+          $innerBlock['attrs']['post_id_to_hydrate_template'] = $attributes['post_id_to_hydrate_template'];
+        }
+        if (isset($innerBlock['blockName'])) {
+          $innerBlocks[] = new Block($innerBlock, $post_id, $post_content, $query_args, $global_theme_styles, $global_db_styles);
+        }
       }
-      $this->innerBlocks = $innerBlocks;
 
-      if($this->name == 'core/gallery'){
-        $classId = $this->get_core_gallery_class_id();
-        $this->inlineClassnames = $classId;
-        $this->inlineStylesheet = $this->get_core_gallery_stylesheet($classId);
+      if (isset($data['attrs']['post_id_to_hydrate_template']) && !empty($data['attrs']['post_id_to_hydrate_template'])) {
+        // set global post object here
+        global $post;
+        $post = get_post($data['attrs']['post_id_to_hydrate_template']);
+        setup_postdata($post);
       }
-    }
 
-    private function get_core_gallery_class_id(){
-      $needle = "wp-block-gallery-";
-      $startPos = strpos($this->dynamicContent, $needle);
-      $endPos = strpos($this->dynamicContent, " ", $startPos);
-      $classId = substr($this->dynamicContent, $startPos, $endPos - $startPos);
-      return $classId;
-    }
-  
-    private function get_core_gallery_stylesheet($classId){		
-      $gap = _wp_array_get( $this->attributes, array( 'style', 'spacing', 'blockGap' ) );
-      // Skip if gap value contains unsupported characters.
-      // Regex for CSS value borrowed from `safecss_filter_attr`, and used here
-      // because we only want to match against the value, not the CSS attribute.
-      if ( is_array( $gap ) ) {
-        foreach ( $gap as $key => $value ) {
-          // Make sure $value is a string to avoid PHP 8.1 deprecation error in preg_match() when the value is null.
-          $value = is_string( $value ) ? $value : '';
-          $value = $value && preg_match( '%[\\\(&=}]|/\*%', $value ) ? null : $value;
-  
-          // Get spacing CSS variable from preset value if provided.
-          if ( is_string( $value ) && str_contains( $value, 'var:preset|spacing|' ) ) {
-            $index_to_splice = strrpos( $value, '|' ) + 1;
-            $slug            = _wp_to_kebab_case( substr( $value, $index_to_splice ) );
-            $value           = "var(--wp--preset--spacing--$slug)";
+
+
+      if ($data['blockName'] === "core/list-item") {
+        $attributes['content'] = substr($htmlContent, strpos($htmlContent, ">") + 1, -5);
+      }
+
+      if ($data['blockName'] == 'core/paragraph') {
+        $attributes['content'] = substr($htmlContent, strpos($htmlContent, ">") + 1, -4);
+        if (isset($attributes['align']) && !empty($attributes['align'])) {
+          $attributes['textAlign'] = $attributes['align'];
+          unset($attributes['align']);
+        }
+      }
+      if ($data['blockName'] == 'core/button') {
+        $dom = new \DOMDocument();
+        // Avoid warnings when loading HTML
+        libxml_use_internal_errors(true);
```

Comment:
Should this be disabled at certain point?

Reply:
We can disable it, but we will have warnings again